karma.conf.js

```diff
@@ -0,0 +1,70 @@
+// Karma configuration
+// Generated on Thu Jun 25 2020 17:12:25 GMT+0200 (Central European Summer Time)
+
+module.exports = function(config) {
+  config.set({
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['jasmine', 'karma-typescript'],
+
+
+    // list of files / patterns to load in the browser
+    files: [
+      'src/**/*.ts',
+      'spec/*Spec.ts'
+    ],
+
+    client: {
+      // leave Jasmine Spec Runner output visible in browser
+      clearContext: false
+    },
+
+    // preprocess matching files before serving them to the browser
+    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+    preprocessors: {
+      "**/*.ts": ['karma-typescript']
+    },
+
+
+    // test results reporter to use
+    // possible values: 'dots', 'progress'
+    // available reporters: https://npmjs.org/browse/keyword/karma-reporter
+    reporters: ['spec', 'progress', 'karma-typescript'],
+
+
+    // web server port
+    port: 9876,
+
+
+    // enable / disable colors in the output (reporters and logs)
+    colors: true,
+
+
+    // level of logging
+    // possible values: config.LOG_DISABLE || config.LOG_ERROR || config.LOG_WARN || config.LOG_INFO || config.LOG_DEBUG
+    logLevel: config.LOG_INFO,
+
+
+    // enable / disable watching file and executing tests whenever any file changes
+    autoWatch: true,
+
+
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['Chrome'],
```

Comment:
We will want the library to work with other browsers. I'd suggest adding Firefox for now as that should be easy and add a TODO comment to test on other browsers. Edge/IE may be hard to test on your Chromebook but they're also more likely to have strange/different behaviour!

Reply:
Once you've resolved a comment you please click 'Resolve conversation' so I can tell when you're done applying changes